chain/x/oracle/module.go

```diff
@@ -49,199 +56,149 @@ func (AppModuleBasic) ValidateGenesis(cdc codec.JSONMarshaler, bz json.RawMessag
-	// }
-
-	// Claim channel capability passed back by IBC module
-	if err := am.keeper.ScopedKeeper.ClaimCapability(ctx, chanCap, ibctypes.ChannelCapabilityPath(portID, channelID)); err != nil {
+	// Claim channel capability passed back by IBC module.
+	capPath := ibctypes.ChannelCapabilityPath(portID, channelID)
+	if err := am.keeper.ScopedKeeper.ClaimCapability(ctx, chanCap, capPath); err != nil {
 		return sdkerrors.Wrap(channel.ErrChannelCapabilityNotFound, err.Error())
 	}
-
 	return nil
 }
 
-func (am AppModule) OnChanOpenAck(
-	ctx sdk.Context,
-	portID,
-	channelID string,
-	counterpartyVersion string,
-) error {
-	// if counterpartyVersion != types.Version {
-	// 	return sdkerrors.Wrapf(porttypes.ErrInvalidPort, "invalid counterparty version: %s, expected %s", counterpartyVersion, "ics20-1")
-	// }
+// OnChanOpenAck implements ics-26 IBCModule interface.
+func (am AppModule) OnChanOpenAck(ctx sdk.Context, portID string, channelID string, counterpartyVersion string) error {
 	return nil
 }
 
-func (am AppModule) OnChanOpenConfirm(
-	ctx sdk.Context,
-	portID,
-	channelID string,
-) error {
+// OnChanOpenConfirm implements ics-26 IBCModule interface.
+func (am AppModule) OnChanOpenConfirm(ctx sdk.Context, portID string, channelID string) error {
 	return nil
 }
 
-func (am AppModule) OnChanCloseInit(
-	ctx sdk.Context,
-	portID,
-	channelID string,
-) error {
+// OnChanCloseInit implements ics-26 IBCModule interface.
+func (am AppModule) OnChanCloseInit(ctx sdk.Context, portID string, channelID string) error {
 	// Disallow user-initiated channel closing for transfer channels
 	return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "user cannot close channel")
 }
 
-func (am AppModule) OnChanCloseConfirm(
-	ctx sdk.Context,
-	portID,
-	channelID string,
-) error {
+// OnChanCloseConfirm implements ics-26 IBCModule interface.
+func (am AppModule) OnChanCloseConfirm(ctx sdk.Context, portID string, channelID string) error {
 	return nil
 }
 
-func (am AppModule) OnRecvPacket(
-	ctx sdk.Context,
-	packet channeltypes.Packet,
-) (*sdk.Result, error) {
-	var requestData OracleRequestPacketData
-	if err := types.ModuleCdc.UnmarshalJSON(packet.GetData(), &requestData); err == nil {
-		calldata, err := hex.DecodeString(requestData.Calldata)
-		if err != nil {
-			return nil, err
-		}
-		// TODO: Mock data source fee payer
-		newMsg := NewMsgRequestData(
-			requestData.OracleScriptID, calldata, requestData.AskCount,
-			requestData.MinCount, requestData.ClientID,
-			sdk.AccAddress([]byte("Unknown")),
-		)
-		return handleMsgRequestDataIBC(
-			ctx, am.keeper, newMsg, packet.GetDestPort(), packet.GetDestChannel(),
-		)
+// OnRecvPacket implements ics-26 IBCModule interface.
+func (am AppModule) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet) (*sdk.Result, error) {
+	var req OracleRequestPacketData
+	if err := types.ModuleCdc.UnmarshalJSON(packet.GetData(), &req); err != nil {
```

Comment:
Note this is the only non-trivial change. I make it return error if message parsing fails.